### ISSUE TITLE
aria-label on input elements reports incorrectly critical issue depending on its type

### ISSUE THREAD

**Issue by rastersysteme** [dequelabs/axe-core#2926]
Product:
axe Extension

Actual:
Using `aria-label` on `input` elements with different used `type` attributes when no `label` is provided reports following error message:
> Elements must only use allowed ARIA attributes

![image](https://user-images.githubusercontent.com/1895414/117863535-2aecd480-b294-11eb-9e1d-36199e3e75ab.png)

Exemplary some input types:

```html
// critical issue
<input type="time" aria-label="some text" />
<input type="password" aria-label="some text" />
<input type="date" aria-label="some text" />

// no problem
<input type="number" aria-label="some text" />
<input type="search" aria-label="some text" />
<input type="text" aria-label="some text" />
```

Expectation:
No critical issue should be reported for **all** input types.

Motivation:
According to [WAI-ARIA 1.1](https://www.w3.org/TR/wai-aria-1.1/#aria-label) and its [practices](https://www.w3.org/TR/wai-aria-practices/#naming_with_aria-label) this behaviour is wrong and should be corrected.

<pre><code>
axe-core version: 4.2.0
axe DevTools - Web Accessibility Testing: 4.9.3

Platform: Mac
Browser: Chrome 90
</code></pre>

**Comment by WilcoFiers** [comment]
Thank you for reporting this. I actually found this exact issue yesterday too and had it on my todo list to report. Axe seems to have a problem determining the implicit role of these elements. Looks like a fairly simple thing to fix.

**Comment by WilcoFiers** [comment]
Forgot to reply, but there are actually two problems here. Firstly that axe-core is failing to identify some input types as `textbox` elements. The second is that global ARIA attributes are failing on elements with no implicit role. Another example of that is:

```html
<canvas aria-label="hello world"></canvas>
```

**Comment by padmavemulapati** [comment]
Verified as part of https://github.com/dequelabs/axe-core/issues/2933 , not reproducing the issue now
